src/Workspaces/Remote/ServiceHub/Services/LanguageServer.cs

```diff
@@ -83,36 +86,52 @@ public Task<VSBeginSymbolParams> BeginWorkspaceSymbolAsync(string query, int sea
                     // for now, we use whatever solution we have currently. in future, we will add an ability to sync VS's current solution
                     // on demand from OOP side
                     // https://github.com/dotnet/roslyn/issues/37424
-                    await SearchAsync(SolutionService.PrimaryWorkspace.CurrentSolution, query, searchId, cancellationToken).ConfigureAwait(false);
-                    return new VSBeginSymbolParams();
+                    var results = await SearchAsync(SolutionService.PrimaryWorkspace.CurrentSolution, args, cancellationToken).ConfigureAwait(false);
+                    return results;
                 }
             }, cancellationToken);
         }
 
-        private async Task SearchAsync(Solution solution, string query, int searchId, CancellationToken cancellationToken)
+        private async Task<SymbolInformation[]> SearchAsync(Solution solution, WorkspaceSymbolParams args, CancellationToken cancellationToken)
         {
-            var tasks = solution.Projects.Select(p => SearchProjectAsync(p, cancellationToken)).ToArray();
-            await Task.WhenAll(tasks).ConfigureAwait(false);
-            return;
-
-            async Task SearchProjectAsync(Project project, CancellationToken cancellationToken)
+            // When progress reporting is supported, report incrementally per project and return an empty result at the end.
+            // Otherwise aggregate and return the results for all projects at the end.
+            if (args.Progress != null)
+            {
+                var tasks = solution.Projects.Select(p => SearchProjectAndReportSymbolsAsync(p, args, cancellationToken)).ToArray();
+                await Task.WhenAll(tasks).ConfigureAwait(false);
```

Comment:
nit: Do we need to use `ToArray` here? It seems likely just passing the IEnumerable would be better.